frontend/packages/kubevirt-plugin/src/components/vm-templates/vm-template-details.tsx

```diff
@@ -0,0 +1,61 @@
+import * as React from 'react';
+
+import { getResource } from 'kubevirt-web-ui-components';
+
+import {
+  Firehose,
+  StatusBox,
+  ScrollToTopOnMount,
+  SectionHeading,
+} from '@console/internal/components/utils';
+
+import { TemplateKind } from '@console/internal/module/k8s';
+import { VMTemplateResourceSummary } from './vm-template-resource';
+import { TemplateModel } from '../../../../../public/models/index';
+
+export const VMTemplateDetailsFirehose = ({ obj: template }: { obj: TemplateKind }) => {
+  const { name, namespace } = template.metadata;
+
+  const vmtRes = getResource(TemplateModel, {
+    name,
+    namespace,
+    isList: false,
+    prop: 'vmt',
+    optional: true,
+  });
+
+  const resources = [vmtRes];
+
+  return (
+    <div className="co-m-pane__body">
+      <Firehose resources={resources}>
```

Comment:
why is the firehose necessary? can't we just use the `obj` which should be the template already? 

Although we will probably need the firehose for passing dataVolumes in the future.

Reply:
> why is the firehose necessary? can't we just use the obj which should be the template already?

IFAIU `getResource` does not get the acutual template ( only do some sanity checks ? ) , so we need to get the template from the store, or k8s API, `Firehose` solves this, what am I missing something ?

Reply:
@suomiy , thanks missed the obj ... :+1:

Reply:
> Although we will probably need the firehose for passing dataVolumes in the future.

Added the dataVolumes to this PR :sun_with_face: